### PR TITLE
Support username/password for cli

### LINKED ISSUE #299: Support --username and --password as parameters to jenkins-cli.jar
When security is enabled, and realm is set to UNIX user/group database, it is impossible to use the jenkins cookbook with a UNIX user. This is because private key authentication is disabled, which in turn is due to [JENKINS-22346](https://issues.jenkins-ci.org/browse/JENKINS-22346).

The `jenkins-cli.jar` utility accepts the parameters `--username` and `--password`, which enables the use of a UNIX user again. This functionality is, sadly, not supported by the cookbook.

I kindly ask that you implement this feature, either as parameters to the resources, or similar to the private key variable in `node.run_state[:jenkins_private_key]`.

### PR DESCRIPTION
**Pull request by simao**
This should fix #299 

Replaces PR #304 With proper branch

### CODE REVIEW

**Comment by chef-supermarket** [comment]
Hi. I am an automated pull request bot named Curry. There are commits in this pull request whose authors are not yet authorized to contribute to Chef Software, Inc. projects or are using a non-GitHub verified email address. To become authorized to contribute, you will need to sign the Contributor License Agreement (CLA) as an individual or on behalf of your company. [You can read more on Chef's blog.](https://www.chef.io/blog/2014/06/23/changes-to-the-contributor-license-agreement-process)

## GitHub Users Who Are Not Authorized To Contribute

The following GitHub users do not appear to have signed a CLA:

* @simao

[Please sign the CLA here.](https://supermarket.chef.io/become-a-contributor)

**Comment by jtimberman** [comment]
@opscode-cookbooks/ociv Please review, this is related to #311.

File: libraries/_executor.rb [hunk 1]
```diff
@@ -68,6 +68,7 @@ def initialize(options = {})
     #
     def execute!(*pieces)
       command_options = pieces.last.is_a?(Hash) ? pieces.pop : {}
+      cli_user, cli_password = options.fetch(:cli_credentials, [])
```
Comment (schisamo) [inline libraries/_executor.rb]: Can we just have two separate options here: `username` and `password`. They should be set via `node.run_state` values in a similar way that the private key is.

**Comment by chef-supermarket** [comment]
Hi. Your friendly Curry bot here. Just letting you know that all commit authors have become authorized to contribute. I have added the "Signed CLA" label to this issue so it can easily be found in the future.

**Comment by schisamo** [comment]
This appears to be a dupe of https://github.com/opscode-cookbooks/jenkins/pull/348.

**Pull request closed without merging**